lib/core/test/unit/Cardano/Pool/DB/Arbitrary.hs

```diff
@@ -133,9 +133,24 @@ arbitraryChainLength = 10
 
 -- NOTE Expected to have a high entropy
 instance Arbitrary PoolId where
-    arbitrary = do
-        bytes <- vector 32
-        return $ PoolId $ B8.pack bytes
+    arbitrary = PoolId . BS.pack <$> vector 32
+    shrink (PoolId p) = do
+        let s = BS.unpack p
+        result <-
+            [ -- Zero out everything:
+              replicate 32 z
+              -- Zero out different halves:
+            ,              replicate 16 z <> drop 16 s
+            , take 16 s <> replicate 16 z
```

Comment:
🆒 

Two just-thoughts:
1. I imagine generalising over [whole, half, quarter, ...] would be possible, if one really wanted to
2. I guess we don't expect the contents of the `PoolId` to ever matter, and such, zeroing out everything _should_ be enough

But mainly:

Would it make sense to move to `Cardano.Wallet.Gen` for re-usability?

Reply:
> I guess we don't expect the contents of the PoolId to ever matter, and such, zeroing out everything should be enough

Unfortunately if we always zero everything out, QuickCheck often fails to find a smaller counterexample, as zeroing out everything is just a little too drastic. :)

If we zero out smaller parts, then we can often still make progress in shrinking towards a smaller counterexample. I did actually manage to find a bug this way (which I eventually fixed).

> I imagine generalising over [whole, half, quarter, ...] would be possible, if one really wanted to

We could indeed! :+1: On this occasion I chose to save the time budget for other things. Though would be interested to see if you can express it in a more concise way?

> Would it make sense to move to Cardano.Wallet.Gen for re-usability?

I agree we could do that. Do you think it would be okay to tackle this in a future task that tackles the duplication among our generators? (I'm trying to limit the scope of this PR as much as possible.)

I think we could actually make a technical debt task for this, as there are many duplicated generators in our source code.